### ISSUE TITLE
[CLI]: Sweep chooses the same parameters each time when method is random

### ISSUE THREAD

**Issue by Leon-Francis** [wandb/wandb#5224]
### Describe the bug

Due to network issues, I deployed wandb to my local server using docker. When I used sweep to tune the parameters, I set the method to random in sweep_config, but the parameter results randomly generated by `wandb.config` were always the same. Setting method to bayes seems have the same issue. Only grid works fine. I have tested that the same code can be used to select parameters randomly in the wandb cloud server, but the parameters are the same every time on the local server.

```python
def main():
    run_name = f'...'
    wandb.init(project='...', name=run_name)
    args = wandb.config

    build_dataset()
    train_process()
    wandb.log({'final_score': final_score})

if __name__ == '__main__':
    sweep_config = {
        'method': 'random',
        'metric': {'goal': 'maximize', 'name': 'final_score'},
        'parameters': {
            'learning_rate': {'values': [1e-3, 1e-4, 1e-5]}
            ...
        }
    }
    sweep_id = wandb.sweep(sweep_config, project='Knowledge Enhanced Stance Detection')
    wandb.agent(sweep_id, main, count=50)
```
There is no exception, only one info. I used ... to mask some personal information.

```
wandb: Currently logged in as: ... Use `wandb login --relogin` to force relogin
Create sweep with ID: 0ny2exor
Sweep URL: ...
2023-03-23_22:11:12 - INFO - pyagent.py           : 316 line - Starting sweep agent: entity=None, project=None, count=50
wandb: Agent Starting Run: f9e2xqyr with config:
wandb:  attention_fusion_learning_rate: 1e-05
wandb:  batch_size: 32
wandb:  cat_add_skip_connection: True
wandb:  cat_bias: True
wandb:  cat_dropout: 0.6
wandb:  cat_structure: {'cat_feature_num_pre_layer': [768, 768], 'cat_head_num_pre_layer': [4]}
wandb:  knowledge_num: 512
wandb:  label_size: 2
wandb:  linear_injection: 512
wandb:  linear_learning_rate: 0.0001
wandb:  max_tokenization_length: 128
wandb:  num_epochs: 1
wandb:  text_with_target: True
wandb:  train_times: 1
wandb:  transformer_learning_rate: 1e-05
wandb:  transformer_name: vinai/bertweet-base
wandb:  transformer_tokenizer_name: vinai/bertweet-base
wandb:  warmup_ratio: 0.1
wandb:  weight_decay: 1e-05
wandb: WARNING Ignored wandb.init() arg project when running a sweep.
wandb: Tracking run with wandb version 0.14.0
wandb: Run data is saved locally in /data/...
wandb: Run `wandb offline` to turn off syncing.
wandb: Syncing run ...
wandb: ⭐️ View project at ...
wandb: 🧹 View sweep at ...
wandb: 🚀 View run at ...
```

### Additional Files

Here are Sweep logs:

Created sweep zuby42it

Created agent tvmfwn3r
Launched new run h83716gg (vital-sweep-1)
Sending commands to agent tvmfwn3r: [{"run_id":"h83716gg","program":"","type":"run","args":{"attention_fusion_learning_rate":{"value":0.00001},"batch_size":{"value":32},"cat_add_skip_connection":{"value":true},"cat_bias":{"value":true},"cat_dropout":{"value":0.2},"cat_structure":{"value":{"cat_feature_num_pre_layer":[768,768],"cat_head_num_pre_layer":[4]}},"knowledge_num":{"value":512},"label_size":{"value":2},"linear_injection":{"value":512},"linear_learning_rate":{"value":0.0001},"max_tokenization_length":{"value":512},"num_epochs":{"value":30},"text_with_target":{"value":true},"train_times":{"value":1},"transformer_learning_rate":{"value":0.00001},"transformer_name":{"value":"roberta-base"},"transformer_tokenizer_name":{"value":"roberta-base"},"warmup_ratio":{"value":0.1},"weight_decay":{"value":0.00001}},"runqueue_item_id":"","logs":[],"run_storage_id":"UnVuOnYxOmg4MzcxNmdnOktub3dsZWRnZSBFbmhhbmNlZCBTdGFuY2UgRGV0ZWN0aW9uOmxlb25fZnJhbmNpcw=="}]
Launched new run nfh3qf3e (spring-sweep-2)
Sending commands to agent tvmfwn3r: [{"run_id":"nfh3qf3e","program":"","type":"run","args":{"attention_fusion_learning_rate":{"value":0.00001},"batch_size":{"value":32},"cat_add_skip_connection":{"value":true},"cat_bias":{"value":true},"cat_dropout":{"value":0.2},"cat_structure":{"value":{"cat_feature_num_pre_layer":[768,768],"cat_head_num_pre_layer":[4]}},"knowledge_num":{"value":512},"label_size":{"value":2},"linear_injection":{"value":512},"linear_learning_rate":{"value":0.0001},"max_tokenization_length":{"value":512},"num_epochs":{"value":30},"text_with_target":{"value":true},"train_times":{"value":1},"transformer_learning_rate":{"value":0.00001},"transformer_name":{"value":"roberta-base"},"transformer_tokenizer_name":{"value":"roberta-base"},"warmup_ratio":{"value":0.1},"weight_decay":{"value":0.00001}},"runqueue_item_id":"","logs":[],"run_storage_id":"UnVuOnYxOm5maDNxZjNlOktub3dsZWRnZSBFbmhhbmNlZCBTdGFuY2UgRGV0ZWN0aW9uOmxlb25fZnJhbmNpcw=="}]
Launched new run w2m1o9cm (lucky-sweep-3)
Sending commands to agent tvmfwn3r: [{"run_id":"w2m1o9cm","program":"","type":"run","args":{"attention_fusion_learning_rate":{"value":0.00001},"batch_size":{"value":32},"cat_add_skip_connection":{"value":true},"cat_bias":{"value":true},"cat_dropout":{"value":0.2},"cat_structure":{"value":{"cat_feature_num_pre_layer":[768,768],"cat_head_num_pre_layer":[4]}},"knowledge_num":{"value":512},"label_size":{"value":2},"linear_injection":{"value":512},"linear_learning_rate":{"value":0.0001},"max_tokenization_length":{"value":512},"num_epochs":{"value":30},"text_with_target":{"value":true},"train_times":{"value":1},"transformer_learning_rate":{"value":0.00001},"transformer_name":{"value":"roberta-base"},"transformer_tokenizer_name":{"value":"roberta-base"},"warmup_ratio":{"value":0.1},"weight_decay":{"value":0.00001}},"runqueue_item_id":"","logs":[],"run_storage_id":"UnVuOnYxOncybTFvOWNtOktub3dsZWRnZSBFbmhhbmNlZCBTdGFuY2UgRGV0ZWN0aW9uOmxlb25fZnJhbmNpcw=="}]
Launched new run o5c4o7lr (royal-sweep-4)
Sending commands to agent tvmfwn3r: [{"run_id":"o5c4o7lr","program":"","type":"run","args":{"attention_fusion_learning_rate":{"value":0.00001},"batch_size":{"value":32},"cat_add_skip_connection":{"value":true},"cat_bias":{"value":true},"cat_dropout":{"value":0.2},"cat_structure":{"value":{"cat_feature_num_pre_layer":[768,768],"cat_head_num_pre_layer":[4]}},"knowledge_num":{"value":512},"label_size":{"value":2},"linear_injection":{"value":512},"linear_learning_rate":{"value":0.0001},"max_tokenization_length":{"value":512},"num_epochs":{"value":30},"text_with_target":{"value":true},"train_times":{"value":1},"transformer_learning_rate":{"value":0.00001},"transformer_name":{"value":"roberta-base"},"transformer_tokenizer_name":{"value":"roberta-base"},"warmup_ratio":{"value":0.1},"weight_decay":{"value":0.00001}},"runqueue_item_id":"","logs":[],"run_storage_id":"UnVuOnYxOm81YzRvN2xyOktub3dsZWRnZSBFbmhhbmNlZCBTdGFuY2UgRGV0ZWN0aW9uOmxlb25fZnJhbmNpcw=="}]
Launched new run jsm4g0th (sweet-sweep-5)
Sending commands to agent tvmfwn3r: [{"run_id":"jsm4g0th","program":"","type":"run","args":{"attention_fusion_learning_rate":{"value":0.00001},"batch_size":{"value":32},"cat_add_skip_connection":{"value":true},"cat_bias":{"value":true},"cat_dropout":{"value":0.2},"cat_structure":{"value":{"cat_feature_num_pre_layer":[768,768],"cat_head_num_pre_layer":[4]}},"knowledge_num":{"value":512},"label_size":{"value":2},"linear_injection":{"value":512},"linear_learning_rate":{"value":0.0001},"max_tokenization_length":{"value":512},"num_epochs":{"value":30},"text_with_target":{"value":true},"train_times":{"value":1},"transformer_learning_rate":{"value":0.00001},"transformer_name":{"value":"roberta-base"},"transformer_tokenizer_name":{"value":"roberta-base"},"warmup_ratio":{"value":0.1},"weight_decay":{"value":0.00001}},"runqueue_item_id":"","logs":[],"run_storage_id":"UnVuOnYxOmpzbTRnMHRoOktub3dsZWRnZSBFbmhhbmNlZCBTdGFuY2UgRGV0ZWN0aW9uOmxlb25fZnJhbmNpcw=="}]
Launched new run fkmthx4e (sweet-sweep-6)
Sending commands to agent tvmfwn3r: [{"run_id":"fkmthx4e","program":"","type":"run","args":{"attention_fusion_learning_rate":{"value":0.00001},"batch_size":{"value":32},"cat_add_skip_connection":{"value":true},"cat_bias":{"value":true},"cat_dropout":{"value":0.2},"cat_structure":{"value":{"cat_feature_num_pre_layer":[768,768],"cat_head_num_pre_layer":[4]}},"knowledge_num":{"value":512},"label_size":{"value":2},"linear_injection":{"value":512},"linear_learning_rate":{"value":0.0001},"max_tokenization_length":{"value":512},"num_epochs":{"value":30},"text_with_target":{"value":true},"train_times":{"value":1},"transformer_learning_rate":{"value":0.00001},"transformer_name":{"value":"roberta-base"},"transformer_tokenizer_name":{"value":"roberta-base"},"warmup_ratio":{"value":0.1},"weight_decay":{"value":0.00001}},"runqueue_item_id":"","logs":[],"run_storage_id":"UnVuOnYxOmZrbXRoeDRlOktub3dsZWRnZSBFbmhhbmNlZCBTdGFuY2UgRGV0ZWN0aW9uOmxlb25fZnJhbmNpcw=="}]
Launched new run eaqyjaqy (stellar-sweep-7)
Sending commands to agent tvmfwn3r: [{"run_id":"eaqyjaqy","program":"","type":"run","args":{"attention_fusion_learning_rate":{"value":0.00001},"batch_size":{"value":32},"cat_add_skip_connection":{"value":true},"cat_bias":{"value":true},"cat_dropout":{"value":0.2},"cat_structure":{"value":{"cat_feature_num_pre_layer":[768,768],"cat_head_num_pre_layer":[4]}},"knowledge_num":{"value":512},"label_size":{"value":2},"linear_injection":{"value":512},"linear_learning_rate":{"value":0.0001},"max_tokenization_length":{"value":512},"num_epochs":{"value":30},"text_with_target":{"value":true},"train_times":{"value":1},"transformer_learning_rate":{"value":0.00001},"transformer_name":{"value":"roberta-base"},"transformer_tokenizer_name":{"value":"roberta-base"},"warmup_ratio":{"value":0.1},"weight_decay":{"value":0.00001}},"runqueue_item_id":"","logs":[],"run_storage_id":"UnVuOnYxOmVhcXlqYXF5Oktub3dsZWRnZSBFbmhhbmNlZCBTdGFuY2UgRGV0ZWN0aW9uOmxlb25fZnJhbmNpcw=="}]
Launched new run hfuie97b (lively-sweep-8)
Sending commands to agent tvmfwn3r: [{"run_id":"hfuie97b","program":"","type":"run","args":{"attention_fusion_learning_rate":{"value":0.00001},"batch_size":{"value":32},"cat_add_skip_connection":{"value":true},"cat_bias":{"value":true},"cat_dropout":{"value":0.2},"cat_structure":{"value":{"cat_feature_num_pre_layer":[768,768],"cat_head_num_pre_layer":[4]}},"knowledge_num":{"value":512},"label_size":{"value":2},"linear_injection":{"value":512},"linear_learning_rate":{"value":0.0001},"max_tokenization_length":{"value":512},"num_epochs":{"value":30},"text_with_target":{"value":true},"train_times":{"value":1},"transformer_learning_rate":{"value":0.00001},"transformer_name":{"value":"roberta-base"},"transformer_tokenizer_name":{"value":"roberta-base"},"warmup_ratio":{"value":0.1},"weight_decay":{"value":0.00001}},"runqueue_item_id":"","logs":[],"run_storage_id":"UnVuOnYxOmhmdWllOTdiOktub3dsZWRnZSBFbmhhbmNlZCBTdGFuY2UgRGV0ZWN0aW9uOmxlb25fZnJhbmNpcw=="}]
Launched new run gl7xkqh4 (lilac-sweep-9)
Sending commands to agent tvmfwn3r: [{"run_id":"gl7xkqh4","program":"","type":"run","args":{"attention_fusion_learning_rate":{"value":0.00001},"batch_size":{"value":32},"cat_add_skip_connection":{"value":true},"cat_bias":{"value":true},"cat_dropout":{"value":0.2},"cat_structure":{"value":{"cat_feature_num_pre_layer":[768,768],"cat_head_num_pre_layer":[4]}},"knowledge_num":{"value":512},"label_size":{"value":2},"linear_injection":{"value":512},"linear_learning_rate":{"value":0.0001},"max_tokenization_length":{"value":512},"num_epochs":{"value":30},"text_with_target":{"value":true},"train_times":{"value":1},"transformer_learning_rate":{"value":0.00001},"transformer_name":{"value":"roberta-base"},"transformer_tokenizer_name":{"value":"roberta-base"},"warmup_ratio":{"value":0.1},"weight_decay":{"value":0.00001}},"runqueue_item_id":"","logs":[],"run_storage_id":"UnVuOnYxOmdsN3hrcWg0Oktub3dsZWRnZSBFbmhhbmNlZCBTdGFuY2UgRGV0ZWN0aW9uOmxlb25fZnJhbmNpcw=="}]
Launched new run 0hw44fbn (different-sweep-10)
Sending commands to agent tvmfwn3r: [{"run_id":"0hw44fbn","program":"","type":"run","args":{"attention_fusion_learning_rate":{"value":0.00001},"batch_size":{"value":32},"cat_add_skip_connection":{"value":true},"cat_bias":{"value":true},"cat_dropout":{"value":0.2},"cat_structure":{"value":{"cat_feature_num_pre_layer":[768,768],"cat_head_num_pre_layer":[4]}},"knowledge_num":{"value":512},"label_size":{"value":2},"linear_injection":{"value":512},"linear_learning_rate":{"value":0.0001},"max_tokenization_length":{"value":512},"num_epochs":{"value":30},"text_with_target":{"value":true},"train_times":{"value":1},"transformer_learning_rate":{"value":0.00001},"transformer_name":{"value":"roberta-base"},"transformer_tokenizer_name":{"value":"roberta-base"},"warmup_ratio":{"value":0.1},"weight_decay":{"value":0.00001}},"runqueue_item_id":"","logs":[],"run_storage_id":"UnVuOnYxOjBodzQ0ZmJuOktub3dsZWRnZSBFbmhhbmNlZCBTdGFuY2UgRGV0ZWN0aW9uOmxlb25fZnJhbmNpcw=="}]
Launched new run q02u3u09 (glowing-sweep-11)

### Environment

WandB version: 0.14.0

OS: 
NAME="CentOS Linux"
VERSION="7 (Core)"
ID="centos"
ID_LIKE="rhel fedora"
VERSION_ID="7"
PRETTY_NAME="CentOS Linux 7 (Core)"
ANSI_COLOR="0;31"
CPE_NAME="cpe:/o:centos:centos:7"

Python version:  3.7.13

Versions of relevant libraries: no relevant libraries

### Additional Context

I noticed that the `runqueue_item_id` in Sweeps logs is an empty str when running in local wandb server. While in cloud server, it has a value like 'UnVuUXVldWVJdGVtOjI3MjkzNDY3Mw=='. I'm wondering if this is the cause of the problem, and if so, how to fix it?

**Comment by Leon-Francis** [comment]
I also tried 0.13 and 0.12 version of WandB, but they also have the same issue.

**Comment by rsanandres-wandb** [comment]
Hello @Leon-Francis! 

Thanks for the context and your code. We have recently received a report similar to this and we are still debugging this issue so thank you for the additional context. I will circle back once we have found a solution to this issue.

**Comment by Leon-Francis** [comment]
Hi @rsanandres-wandb,

Really appreciate for your time and attention❤. Let me know if you need more logs related to this problem.

**Comment by gtarpenning** [comment]
Bug identified and release impending... Thanks for the report @Leon-Francis :)  If this issue is time-sensitive, and in the future, you can (very likely just) downgrade to the previous release version. In this case: 0.29.0.

PS: run queues are handled differently in cloud vs. local server for performance reasons, which is why you might see empty `runqueue_item_id` if you run the server locally.

**Comment by luckyhug** [comment]
I found that the sweep agents' parameters are not passing to the training script. I set my sweep epochs min:320, max:600 to avoid the default training script settings. But every time the training epochs is set to 300, which is just the same as in the default training script:

(group.add_argument('--epochs', type=int, default=300, metavar='N',
help='number of epochs to train (default: 300)'))

**Comment by Leon-Francis** [comment]
Hi @gtarpenning,

Thank you and the wandb team so much for your fix. I have noticed that the wandb docker published 0.30.1 . After updating to this version, this bug has been resolved. I will close this issue.
